com.microsoft.mrtk.graphicstools.unity/Runtime/MeshOutline/MeshOutlineHierarchy.cs

```diff
@@ -21,15 +22,18 @@ public class MeshOutlineHierarchy : BaseMeshOutline
         /// </summary>
         private void Awake()
         {
-            MeshRenderer[] meshRenderers = GetComponentsInChildren<MeshRenderer>();
-            meshOutlines = new MeshOutline[meshRenderers.Length];
+            Renderer[] meshRenderers = GetComponentsInChildren<Renderer>();
+            meshOutlines = new List<MeshOutline>();
 
             for (int i = 0; i < meshRenderers.Length; ++i)
             {
-                var meshOutline = meshRenderers[i].gameObject.AddComponent<MeshOutline>();
-                meshOutline.OutlineMaterial = outlineMaterial;
-                meshOutline.OutlineWidth = outlineWidth;
-                meshOutlines[i] = meshOutline;
+                if (meshRenderers[i].GetComponent<SpriteRenderer>() == null)

```

Comment:
Rather than calling GetComponent could you do `if (meshRenderers[i] as SpriteRenderer == null)`?

Reply:
great callout